src/Microsoft.Tye.Core/DeployApplicationKubernetesManifestStep.cs

```diff
@@ -59,6 +60,47 @@ public override async Task ExecuteAsync(OutputContext output, ApplicationBuilder
             }
 
             output.WriteInfoLine($"Deployed application '{application.Name}'.");
+            if (application.Ingress.Count > 0)
+            {
+                output.WriteInfoLine($"Waiting for ingress to be deployed. This may take a long time.");
+                foreach (var ingress in application.Ingress)
+                {
+                    using var ingressStep = output.BeginStep($"Retrieving details for {ingress.Name}...");
+
+                    var done = false;
+
+                    Action<string> complete = line =>
+                    {
+                        done = line != "''";
+                        if (done)
+                        {
+                            output.WriteInfoLine($"IngressIP: {line}");
+                        }
+                    };
+
+                    var retries = 0;
+                    while (!done && retries < 60)
+                    {
+                        var ingressExitCode = await Process.ExecuteAsync(
+                            "kubectl",
+                            $"get ingress {ingress.Name} -o jsonpath='{{..ip}}'",
+                            Environment.CurrentDirectory,
+                            complete,
+                            capture.StdErr);
+
+                        if (ingressExitCode != 0)
+                        {
+                            throw new CommandException("'kubectl get ingress' failed");
+                        }
+
+                        if (!done)
+                        {
+                            await Task.Delay(2000);
```

Comment:
Sure we want to wait 2 seconds? That seems quite long, especially if there are 60 retries.

Reply:
It takes at least 20 seconds from my trials